pandas/core/indexes/base.py

```diff
@@ -100,7 +100,12 @@
 )
 
 if TYPE_CHECKING:
+    from typing import final
```

Comment:
can you instead define this in _typing?

Reply:
huh, i thought i couldnt bc of the static-checking thing, but it looks like i can.  will update